### ISSUE TITLE
EXTENDED_ ARG instruction not supported

### ISSUE THREAD

**Issue by hasikill** [extremecoders-re/bytecode_simplifier#9]
Exception: EXTENDED_ARG not yet implemented

**Comment by abel1502** [comment]
Duplicate of #2.

Brielfy: I've fixed it [here](https://github.com/abel1502/bytecode_simplifier), but those changes weren't yet incorporated into the original repo

**Comment by AGPonomarev** [comment]
> Duplicate of #2.
> 
> Brielfy: I've fixed it [here](https://github.com/abel1502/bytecode_simplifier), but those changes weren't yet incorporated into the original repo

Hello, did you manage to unpack the mod WoT?

**Comment by abel1502** [comment]
> > Duplicate of #2.
> > Brielfy: I've fixed it [here](https://github.com/abel1502/bytecode_simplifier), but those changes weren't yet incorporated into the original repo
> 
> Hello, did you manage to unpack the mod WoT?

I actually found out that I had been dealing with a combined use of several obfuscators, so this tool wasn't sufficient. I ended up extracting the mod from the imported modules cache directly, via another mod

**Comment by AGPonomarev** [comment]
> I ended up extracting the mod from the imported modules cache directly, via another mod

Is there a guide or article on how to do this?

**Comment by abel1502** [comment]
> > I ended up extracting the mod from the imported modules cache directly, via another mod
> 
> Is there a guide or article on how to do this?

Nope, I don't think there is. At least I don't know of one. I just experimented with what I can get from inside the same pyton interpreter as the mod. If you'd like to have a go at this yourself, good points to get started at are `sys.modules` and `gc.get_objects()`. Another strategy is to overwrite some library functions/classes and then go up the traceback via `inspect`'s methods. I'm sorry, but I don't have spare time in the foreseeable future to get into greater details or write a tutorial, but I had some plans to encapsulate it into some sort of toolkit. Don't expect it to be soon neither, though

UPD: Also, I just came up with another idea of using `sys.addaudithook()` for this cause. It might actually be the simplest one, now that I think of it
